### PR TITLE
Fully migrate from travis to github actions

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -28,7 +28,9 @@ jobs:
       - name: Code style check and binary-compatibility check
         run: |-
           cp .jvmopts-travis .jvmopts
-          sbt scalafmtCheckAll scalafmtSbtCheck headerCheckAll grpcVersionSyncCheck +mimaReportBinaryIssues
+          sbt scalafmtCheckAll scalafmtSbtCheck headerCheckAll grpcVersionSyncCheck +mimaReportBinaryIssues whitesourceCheckPolicies
```
Comment (jrudolph) [inline .github/workflows/build-test.yml:31]: Run whitesourceCheck also on PRs and not only after publishing in `main`.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -29,8 +31,31 @@ jobs:
           SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
           SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
 
+      - name: Update Whitesource Listing
+        run: sbt whitesourceUpdate
+        env:
+          WHITESOURCE_PASSWORD: "${{ secrets.WHITESOURCE_PASSWORD }}"
+
   # TODO publish gradle from here as well
```
Comment (raboof) [inline .github/workflows/publish.yml:39]: We can drop the TODO now ;)

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -29,8 +31,31 @@ jobs:
...
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+
+      - name: Set up JDK 8
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.8
+
+      - name: Publish Plugin to Gradle Plugin Repository
+        run: cd gradle-plugin && ./gradlew publishPlugins -Pgradle.publish.key='${{ secrets.GRADLE_PUBLISH_KEY }}' -Pgradle.publish.secret='${{ secrets.GRADLE_SECRET }}'
```
Comment (raboof) [inline .github/workflows/publish.yml:57]: I wonder why we didn't do it like this on the travis side before. :+1:

**Review by raboof — APPROVED** [review]
really great. we can remove that TODO now ;)

File: .github/workflows/publish.yml (hunk 3 above)
Comment (jrudolph) [inline .github/workflows/publish.yml:57]: Does the gradle plugin itself depend on anything from the sbt build? Or only the builds using it? So, would we have to publish the sbt artifacts before?

Comment (jrudolph) [reply .github/workflows/publish.yml:57]: Should probably work, as it's the same as in the existing travis build.

Comment (jrudolph) [reply .github/workflows/publish.yml:57]: Yep, don't know I found it in the docs.

**Pull request merged**